### PR TITLE
Implemented Self Attention for Question Encoding

### PR DESCRIPTION
**Pull request by Drajan**
Multi-head Attention weights are computed using two FF layers with ReLU activation and softmax for probabilities ==> softmax(FF(ReLU(FF(input))))
input = [batch_size, num_words, embed_size]
attention = [batch_size, num_words, num_attention_heads]
input_attention_weighted = [batch_size, num_heads, embed_size]
output = [batch_size, num_heads*embed_size] ==> concatenating multi-head representation

### CODE REVIEW

**Comment by tkornut** [comment]
Early results:
 * 7 GPUs enabled me to use batch size of 300
 * on that setting, one epoch (i.e. processing of 4 folds of joined training and validation sets, ~3000 samples) takes around 30 seconds

Screen Shot 2019-05-03 at 4 31 06 PM
